roles.cpp

```diff
@@ -0,0 +1,163 @@
+    "roleOpPreDecrement",
+    "roleOpPostDecrement",
+    "roleOpNegative",
+    "roleOpPositive",
+    "roleOpBitwiseComplement",
+    "roleOpDereference",
+    "roleOpTakeAddress",
+    "roleFile",
+    "roleOpBooleanAnd",
+    "roleOpBooleanOr",
+    "roleOpBooleanNot",
+    "roleOpBooleanXor",
+    "roleOpAdd",
+    "roleOpSubstract",
+    "roleOpMultiply",
+    "roleOpDivide",
+    "roleOpMod",
+    "rolePackageDeclaration",
+    "roleImportDeclaration",
+    "roleImportPath",
+    "roleImportAlias",
+    "roleFunctionDeclaration",
+    "roleFunctionDeclarationBody",
+    "roleFunctionDeclarationName",
+    "roleFunctionDeclarationReceiver",
+    "roleFunctionDeclarationArgument",
+    "roleFunctionDeclarationArgumentName",
+    "roleFunctionDeclarationArgumentDefaultValue",
+    "roleFunctionDeclarationVarArgsList",
+    "roleTypeDeclaration",
+    "roleTypeDeclarationBody",
+    "roleTypeDeclarationBases",
+    "roleTypeDeclarationImplements",
+    "roleVisibleFromInstance",
+    "roleVisibleFromType",
+    "roleVisibleFromSubtype",
+    "roleVisibleFromPackage",
+    "roleVisibleFromSubpackage",
+    "roleVisibleFromModule",
+    "roleVisibleFromFriend",
+    "roleVisibleFromWorld",
+    "roleIf",
+    "roleIfCondition",
+    "roleIfBody",
+    "roleIfElse",
+    "roleSwitch",
+    "roleSwitchCase",
+    "roleSwitchCaseCondition",
+    "roleSwitchCaseBody",
+    "roleSwitchDefault",
+    "roleFor",
+    "roleForInit",
+    "roleForExpression",
+    "roleForUpdate",
+    "roleForBody",
+    "roleForEach",
+    "roleWhile",
+    "roleWhileCondition",
+    "roleWhileBody",
+    "roleDoWhile",
+    "roleDoWhileCondition",
+    "roleDoWhileBody",
+    "roleBreak",
+    "roleContinue",
+    "roleGoto",
+    "roleBlock",
+    "roleBlockScope",
+    "roleReturn",
+    "roleTry",
+    "roleTryBody",
+    "roleTryCatch",
+    "roleTryFinally",
+    "roleThrow",
+    "roleAssert",
+    "roleCall",
+    "roleCallReceiver",
+    "roleCallCallee",
+    "roleCallPositionalArgument",
+    "roleCallNamedArgument",
+    "roleCallNamedArgumentName",
+    "roleCallNamedArgumentValue",
+    "roleNoop",
+    "roleBooleanLiteral",
+    "roleByteLiteral",
+    "roleByteStringLiteral",
+    "roleCharacterLiteral",
+    "roleListLiteral",
+    "roleMapLiteral",
+    "roleNullLiteral",
+    "roleNumberLiteral",
+    "roleRegexpLiteral",
+    "roleSetLiteral",
+    "roleStringLiteral",
+    "roleTupleLiteral",
+    "roleTypeLiteral",
+    "roleOtherLiteral",
+    "roleMapEntry",
+    "roleMapKey",
+    "roleMapValue",
+    "roleType",
+    "rolePrimitiveType",
+    "roleAssignment",
+    "roleAssignmentVariable",
+    "roleAssignmentValue",
+    "roleAugmentedAssignment",
+    "roleAugmentedAssignmentOperator",
+    "roleAugmentedAssignmentVariable",
+    "roleAugmentedAssignmentValue",
+    "roleThis",
+    "roleComment",
+    "roleDocumentation",
+    "roleWhitespace",
+};
+#define TOTAL_ROLES  141
+
+const char *role_name_for_id(uint16_t id) {
```

Comment:
We should formalize a code style and follow it. There's currently a mix of style with curly braces:
```
fn () {
  if () {
  }
}
```
vs
```
fn ()
{
  if ()
  {
  }
}
```
and also with spacing (i.e.: `if(id == 0)` vs `if (__V__ == NULL)`). 

I suggest [Google's C++ style guide](https://google.github.io/styleguide/cppguide.html)